TestBenches/InputRouter_test.cpp

```diff
@@ -230,82 +396,155 @@ void prepareInputStreams( ifstream * pInputStreams
     // get phi bin 
     ap_uint<kSizeBinWord> hBnWrd = hPhBnWord.range(kSizeBinWord * cLyrIndx + (kSizeBinWord-1), kSizeBinWord * cLyrIndx);
     auto cNPhiBns =  (1+(unsigned int)(hBnWrd));
-    std::cout << "Layer# " << cLyrIndx 
+    std::cout << "Layer# " << hLyrId 
+      << "[Index=" << hLyrId << "]"
       << " " << cNPhiBns 
       << " phi bins"
+      << " barrel bit is set to "
+      << +hIsBrl
       << "\n";
-    
+    int cLyrCorr = (hIsBrl==1) ? 0 : 10; 
     for( int cPhiBn=0; cPhiBn<cNPhiBns; cPhiBn++)
     {
-      std::string cMemPrint = getMemPrintName(cDTCName ,cLyrIndx, cPhiBn, pNonant, hLinkWord);
-      if( !cMemPrint.empty() )
-      {
-        std::cout << "Loading stubs from .. " << cMemPrint << "\n";
-        openDataFile(pInputStreams[cMemIndx],cMemPrint); 
+      std::string cFileName = "emData/MemPrints/InputStubs/"; 
+      std::string cMemPrint = getMemPrintName( cDtcName, pDTCsplit, pNonant, hLyrId + cLyrCorr, cPhiBn ); 
+      cFileName += cMemPrint;
+
+      std::ifstream fin_mem_prints;
+      bool cPrintError=false;
+      bool cFileFnd = openDataFile(fin_mem_prints,cFileName,cPrintError);
+      if( cFileFnd ){ 
+        fin_mem_prints.close();
+        std::cout << "\t..Loading stubs from .. " << cFileName 
+          << " [Mem# " << +cMemIndx << "]\n";
+        openDataFile(pInputStreams[cMemIndx],cFileName); 
+        cMemIndx++;
       }
-      cMemIndx++;
     }
   }
 }
 
+// for now .. create LUT for IR 
+// this can eventually be produced 
+// by either the wiring script 
+// or the emulation 
+void prepareLUTs( int pDTCsplit = 0
```

Comment:
@sseifeln How is the LUT created by this function used? It seems like the test-bench creates this file, but then doesn't use it unless you uncomment some lines.

Reply:
I can't remember now but will have a look first thing Monday (and also fix the other request!)

Reply:
I remember now :) I think I was toying around with the idea of having a LUT for how many memories are filled by each link - and then didn't follow it through. Will remove the code from the test bench.